gobblin-compliance/src/main/java/gobblin/compliance/HivePurgerWriter.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.compliance;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.writer.DataWriter;
+
+/**
+ * This class is responsible for executing all purge queries and altering the partition location if the original
+ * partition is not modified during the current execution.
+ *
+ * @author adsharma
+ */
+@Slf4j
+@AllArgsConstructor
+public class HivePurgerWriter implements DataWriter<HivePurgerPartitionRecord> {
+  private FileSystem fs;
+  private HivePurgerQueryExecutor hivePurgerQueryExecutor;
+
+  /**
+   * This method is responsible for actual purging.
```

Comment:
Please add some doc.
1. What is the staging logic?
2. How do we ensure atomicity?
3. What happens in `write(record)` and what happens in `commit()`?
4. What is the commit criteria?
5. If commit fails will staging table/data be cleaned up?